src/library/scala/collection/immutable/ArraySeq.scala

```diff
@@ -303,6 +303,26 @@ object ArraySeq extends StrictOptimizedClassTagSeqFactory[ArraySeq] { self =>
         AnyStepper.ofParIntStepper(new WidenedByteArrayStepper(unsafeArray, 0, unsafeArray.length))
       else new WidenedByteArrayStepper(unsafeArray, 0, unsafeArray.length)
     ).asInstanceOf[S with EfficientSplit]
+    override def updated[B >: Byte](index: Int, elem: B): ArraySeq[B] =
+      elem match {
+        case b: Byte =>
+          val array = new Array[Byte](unsafeArray.length)
+          System.arraycopy(unsafeArray, 0, array, 0, unsafeArray.length)
+          array(index) = b
+          new ArraySeq.ofByte(array)
```

Comment:
Is there a reason you didn't use `case b: Byte => new ArraySeq.ofByte(unsafeArray.updated(index, b))` here?

Reply:
Silly mistake. Sorry about that. Fixed.